Dockerfile

```diff
@@ -1,49 +1,52 @@
-FROM debian:buster-slim
+### Electrum Rust Server ###
+FROM rust:1.41.1-slim as electrs-build
 RUN apt-get update
+RUN apt-get install -qq -y clang cmake
+RUN rustup component add rustfmt
 
-### Bitcoin ###
-
-# Buildtime dependencies
-RUN apt-get install -qqy build-essential libtool autotools-dev automake pkg-config bsdmainutils python3
+# Build, test and install electrs
+WORKDIR /build/electrs
+COPY . .
+RUN cargo fmt -- --check
+RUN cargo build --locked --release --all
+RUN cargo test --locked --release --all
 
-# Runtime dependencies
+FROM debian:buster-slim as updated
+RUN apt-get update
+# Install Bitcoin Core runtime dependencies
 RUN apt-get install -qqy libevent-dev libboost-system-dev libboost-filesystem-dev libboost-test-dev libboost-thread-dev
+# Install a few test tools
+RUN apt-get install -qqy jq netcat
```

Comment:
AFAIK merging apt calls together should make it faster, am I missing something?

Reply:
You're right - good catch :)
Fixed in b20a94c.